INICIALIZAÇÃO DO PROJETO

BrunoSantos98 opened the pull request:
Criação da entidade, rota de cadastro, rota de consulta por ID e exceptions

BrunoSantos98 committed:
Criação da entidade person, exceptions e duas rotas uma de cadastro e uma de consulta por id

BrunoSantos98 committed:
COrreção de Stakcs para stack

Andrewsoares15 commented on spring-boot/src/main/java/com/people/peoplemanager/api/PeopleController.java:
```diff
@@ -1,16 +1,72 @@
+    }
+
+    private void verificacaoString(String text, int tam, String namefield){
+        if(text.isEmpty() || text.split("").equals("") && namefield.length() != 6){
```
Mano, ficou dhr essa implementação que tu fez. No spring conseguimos fazer essa validação utilizando anotações também, se tiver interesse da uma olhada: https://www.bezkoder.com/spring-boot-validate-request-body/

BrunoSantos98 replied:
sim, usando validation é massa demais, ja usei outras vezes mas não sei se poderia colocar aqui no nosso projeto kkk por isso nao coloquei validation e lombock por exemplo, mas caso possa eu coloco e ja arrumo essas validaçoes

BrunoSantos98 committed:
Correção de algumas regras do controller, falta somente implementar a regra do recebimento da data

andrepaschoal commented on spring-boot/pom.xml:
```diff
@@ -22,6 +22,16 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-web</artifactId>
 		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>org.postgresql</groupId>
+			<artifactId>postgresql</artifactId>
+			<scope>runtime</scope>
+		</dependency>
```
Já que estamos adicionando dependência para um BD, acho que faz sentido termos uma maneira fácil de iniciar um servidor Postgres a partir do projeto. Um docker-compose eria interessante 👍

andrepaschoal commented on spring-boot/src/main/java/com/people/peoplemanager/repository/PersonRepository.java:
```diff
@@ -0,0 +1,13 @@
+import java.util.UUID;
+
+@Repository
+public interface PersonRepository extends JpaRepository<Person, UUID> {
```
Pra ficar consistente com o Controller
```suggestion
public interface PeopleRepository extends JpaRepository<Person, UUID> {
```

andrepaschoal commented on spring-boot/src/main/java/com/people/peoplemanager/model/Person.java:
```diff
@@ -0,0 +1,74 @@
+
+@Entity
+@Table(name = "PERSON_TB")
+public class Person {
```
Será que também poderia ser um `record`?

BrunoSantos98 replied:
Eu tentei primeiro fazer e não rolou, fiz uma pesquisa rapida e encontrei isso: https://thorben-janssen.com/java-records-hibernate-jpa/#records-cant-be-entities. No caso acredito que a classe Person não terá como se tornar um record certo?

andrepaschoal replied:
é, não dá não

andrepaschoal commented on spring-boot/src/main/java/com/people/peoplemanager/model/Person.java:
```diff
@@ -0,0 +1,74 @@
+    private String nome;
+    @Column(nullable = false, length = 10)
+    private String nascimento;
+    private String[] stack;
```
Eu adicionaria um `TODO` ou `FIXME` nessa linha pra ficar claro que este campo não está sendo persistido.

BrunoSantos98 replied:
eu adicionei no meu codigo, em commits posteriores, será se eu nao mandei?

andrepaschoal commented on spring-boot/src/main/java/com/people/peoplemanager/model/Person.java:
```diff
@@ -0,0 +1,74 @@
+        return stack;
+    }
+
+    public void setStacks(String[] stacks) {
+        this.stack = stacks;
```
```suggestion
    public void setStacks(String[] stack) {
        this.stack = stack;
```

andrepaschoal commented on spring-boot/src/main/java/com/people/peoplemanager/model/Person.java:
```diff
@@ -0,0 +1,74 @@
+    @Column(nullable = false, length = 100)
+    private String nome;
+    @Column(nullable = false, length = 10)
+    private String nascimento;
```
Ia sugerir usar `Instant`, mas `String` tá de bom tamanho. Acho que não tem nenhuma regra associada a essa data, né?

BrunoSantos98 replied:
Eu pensei em usar um LocalDate ou até mesmo um Date, que sei la eu acho mais correto, porem acredito que isso poderia aumentar a complexidade do projeto, não sei

andrepaschoal replied:
Sobre aumentar a complexidade: na verdade simplifica. e.g. Um request com uma data inválida falha antes de chegar no controller e não é necessário nenhum código pra validar formato.

BrunoSantos98 replied:
Então nesse caso você indicaria qual tipo de data?

andrepaschoal commented on spring-boot/src/main/java/com/people/peoplemanager/model/Person.java:
```diff
@@ -0,0 +1,74 @@
+
+import jakarta.persistence.*;
+
+import java.util.List;
```
```suggestion
```

andrepaschoal commented on spring-boot/src/main/java/com/people/peoplemanager/exceptions/handlerExceptions/StandardError.java:
```diff
@@ -0,0 +1,63 @@
+
+import java.time.Instant;
+
+public class StandardError {
```
Mais um que poderia ser record? 🤷

andrepaschoal commented on spring-boot/src/main/java/com/people/peoplemanager/exceptions/ObjectNotFoundException.java:
```diff
@@ -0,0 +1,7 @@
+package com.people.peoplemanager.exceptions;
+
+public class ObjectNotFoundException extends RuntimeException{
```
```suggestion
public class PersonNotFoundException extends RuntimeException{
```

andrepaschoal commented on spring-boot/src/main/java/com/people/peoplemanager/exceptions/handlerExceptions/ServiceExceptionHandler.java:
```diff
@@ -0,0 +1,39 @@
+
+    @ExceptionHandler(InvalidObjectException.class)
+    public ResponseEntity<StandardError> InvalidObjectException(InvalidObjectException e, HttpServletRequest request){
+        StandardError error = new StandardError(
+                Instant.now(),
+                HttpStatus.UNPROCESSABLE_ENTITY.value(),
+                "Requisição invalida por conta de objeto(s) não aceito(s)",
+                e.getMessage(),
+                request.getRequestURI()
+        );
+        return ResponseEntity.status(HttpStatus.UNPROCESSABLE_ENTITY).body(error);
```
Me parece que não há requisito com relação ao corpo da mensagem de erro né? Nesse caso eu não enviaria mensagem alguma, apenas se o Spring pedir.
```suggestion
        return ResponseEntity.status(HttpStatus.UNPROCESSABLE_ENTITY);
```

BrunoSantos98 replied:
Mas e no caso as exceptions? eu herdo elas de Runtime Exception com construtor pegando metodo de inserir mensagem do Runtime. Nesse caso nas Exceptions eu tiraria esse contruto ou deixo?

andrepaschoal replied:
As mensagens nas exceções não são necessárias, acho que podem até ter apenas construtor sem argumentos.

andrepaschoal reviewed (changes requested):
Com a dependência do banco o servidor não vai mais subir. Faz sentido adicionar uma forma fácil de executar o banco de dados, um docker-compose da vida.

BrunoSantos98 committed:
Adicionando uma nomenclatura consistente com o controller

Co-authored-by: andrepaschoal <paschoal.andre@gmail.com>

BrunoSantos98 committed:
Correção nome de classe e outras nomenclaturas

Alem da correção do nome da classe solicitada, existe a correção do nome da exception que foi alterada de ObjectNotFound para PersonNotFound, alem desta tem a correção do nome de repository que antes era PersonRepository e agora se chama PeopleRepository

BrunoSantos98 committed:
ALteração de class para record

BrunoSantos98 committed:
ALteração no nome de ObjectNotFoundException para PersonNotFoundException

BrunoSantos98 committed:
Inserção do Fixme para atributo nao mapeado e correção dos nomes stacks restantes para stack

BrunoSantos98 committed:
Correção EXceptions

Correção das exceptions para constrrutores vazios e correção do uso das mesmas na classe controllerrr
